framework/include/base/MooseApp.h

```diff
@@ -468,6 +468,12 @@ class MooseApp :
   /// Returns whether the Application is running in check input mode
   bool checkInput() const { return _check_input; }
 
+  /// Sets a Boolean indicating that at least one object is requesting performance logging in this application
+  void setLoggingRequested() { _logging_requested = true; }
+
+  /// Returns a Boolean indicating whether performance logging is requested in this application
+  bool getLoggingRequested() const { return _logging_requested; }
+
```

Comment:
Just to reduce adding more stuff to MooseApp, should these just go into OutputWarehouse? The Console and PerformanceData have access to the OutputWarehouse from _app, but if you prefer this I don't mind that much.